crates/omnix-health/src/check/homebrew.rs

```diff
@@ -0,0 +1,169 @@
+use nix_rs::{flake::url::FlakeUrl, info::NixInfo};
+use serde::{Deserialize, Serialize};
+use std::process::Command;
+
+use crate::traits::{Check, CheckResult, Checkable};
+
+/// Check if Homebrew is installed
+#[derive(Debug, Serialize, Deserialize, Clone)]
+#[serde(default, rename_all = "kebab-case")]
+pub struct Homebrew {
+    pub(crate) enable: bool,
+    /// Whether to produce [Check::required] checks
+    pub(crate) required: bool,
+    /// Whether to warn if Homebrew is installed (default: true on macOS)
+    pub(crate) warn_if_installed: bool,
+}
+
+impl Default for Homebrew {
+    fn default() -> Self {
+        Self {
+            enable: true,
+            required: false,
+            warn_if_installed: true,
+        }
+    }
+}
+
+impl Checkable for Homebrew {
+    fn check(
+        &self,
+        nix_info: &NixInfo,
+        _flake_url: Option<&FlakeUrl>,
+    ) -> Vec<(&'static str, Check)> {
+        let mut checks = vec![];
+        if !self.enable {
+            return checks;
+        }
+
+        // Only check on macOS by default, but allow override
+        if !matches!(nix_info.nix_env.os, nix_rs::env::OS::MacOS { .. }) && self.warn_if_installed {
+            return checks;
+        }
+
+        let homebrew_install_result = detect_homebrew();
+        checks.push((
+            "homebrew-check",
+            installation_check(
+                &homebrew_install_result,
+                self.required,
+                self.warn_if_installed,
+            ),
+        ));
+
+        checks
+    }
+}
+
+/// Result of Homebrew detection
+#[derive(Debug)]
+pub struct HomebrewInstall {
+    pub bin_path: std::path::PathBuf,
+    pub version: String,
```

Comment:
We don't strictly need to show the version. Let's remove it. Less code is better.